File: leetcode/easy/Q101_SymmetricTree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        que = [root, root]
        while que:
            node1 = que.pop(0)
            node2 = que.pop(0)
            if node1 is None and node2 is None:
                continue
            if node1 is None or node2 is None:
                return False
            if node1.val != node2.val:
                return False
            que.append(node1.left)
            que.append(node2.right)
            que.append(node1.right)
            que.append(node2.left)
        return True

File: src/easy/Q101_SymmetricTree.java
package easy;

public class Q101_SymmetricTree {

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
        }
    }
    class Solution {
        public boolean isSymmetric(TreeNode root) {
           return isSameTree(root.left, root.right);
        }
        public boolean isSameTree(TreeNode left, TreeNode right){
            if(left == null && right == null) return true;
            if(left == null && right !=null) return false;
            if(left != null && right == null) return false;

            if(left.val == right.val){
                return isSameTree(left.right, right.left) && isSameTree(left.left, right.right);
            }else{
                return false;
            }
        }
    }
}
